src/ASM/ASMs2D.C

```diff
@@ -3068,3 +3068,25 @@ size_t ASMs2D::getNoProjectionNodes () const
 {
   return proj->numCoefs_u() * proj->numCoefs_v();
 }
+
+
+void ASMs2D::getNeighbours (NeighArray& neigh) const
+{
+  int N1, N2, dummy;
+  this->getNoStructElms(N1, N2, dummy);
+  for (int i = 0; i < static_cast<int>(this->getNoElms(true)); ++i) {
```

Comment:
I think maybe a double loop over parameter directions (like in `genereteFEMTopology`) is preferable. Then this can be done without all the if- and mod-ery below. At least in 3D that will create more readable code.

Reply:
i don't immediately see how a double loop would help.